installinator/src/write.rs

```diff
@@ -759,21 +768,56 @@ impl ControlPlaneZoneWriteContext<'_> {
             .register();
     }
 
-    fn mupdate_override_artifact(
+    async fn mupdate_override_artifact(
         &self,
         mupdate_uuid: MupdateOverrideUuid,
-    ) -> BufList {
-        // Might be worth writing out individual hash IDs for each zone in the
-        // future.
+    ) -> Result<BufList, JoinError> {
         let hash_ids =
             [self.host_phase_2_id.clone(), self.control_plane_id.clone()]
                 .into_iter()
                 .collect();
-        let mupdate_override = MupdateOverrideInfo { mupdate_uuid, hash_ids };
+        let zones = compute_zone_hashes(&self.zones).await?;
+
+        let mupdate_override =
+            MupdateOverrideInfo { mupdate_uuid, hash_ids, zones };
         let json_bytes = serde_json::to_vec(&mupdate_override)
             .expect("this serialization is infallible");
-        BufList::from(json_bytes)
+        Ok(BufList::from(json_bytes))
+    }
+}
+
+/// Computes the zone hash IDs.
+///
+/// Hash computation is done in parallel on blocking tasks. If a task panics
+/// (should not happen in normal use), a `JoinError` is returned.
+async fn compute_zone_hashes(
+    images: &ControlPlaneZoneImages,
+) -> Result<IdMap<MupdateOverrideZone>, JoinError> {
+    let mut tasks = images
+        .zones
+        .iter()
+        .map(|(file_name, data)| {
+            let file_name = file_name.clone();
+            // data is a Bytes so is cheap to clone.
+            let data: Bytes = data.clone();
+            // Compute hashes in parallel.
```

Comment:
Should this use `JoinSet::spawn_blocking()` since this work is synchronous and compute bound? (I assume `collect()` does the equivalent of `JoinSet::spawn()`?)

Reply:
It should, I started off with that but got too clever by half, whoops. Fixed.